### ISSUE TITLE
[Feature Request] Mission Mode: Terrain Follow based on rangefinder data

### ISSUE THREAD

**Issue by FlavioTonelli** [PX4/Firmware#15411]
During some activities on the field we met the need for an autonomous flight at constant altitude from the terrain. 
What PX4 can right now do is accept a mission expressed with ASML altitude and execute that mission. So a basic terrain follow mission can right now be achieved exploiting the map terrain data during the planning phase.
This approach however suffer on accuracy and can not be used when a terrain constant altitude is critical(such as in low altitude flight).
Implement a mission mode based on rangefinder data could be one solution to improve the performance of the firmware in that real world scenario.

**Comment by julianoes** [comment]
@reedev that's what you want as well, right

**Comment by reedev** [comment]
Yes, partly. It's not about flying over the terrain (although this is the safest way). My goal would be to get to a waypoint at a certain distance from the terrain.

**Comment by julianoes** [comment]
Right, but based on rangefinder altitude?

**Comment by reedev** [comment]
Yes indeed

**Comment by dakejahl** [comment]
This doesn't already work? Am I understanding correctly - missions cannot be defined as terrain following? Or do you just want specific waypoints to use distance from ground, but not the entire mission?

**Comment by reedev** [comment]
Good question. If I remember correctly I have seen a video one day from someone you was following a terrain on some hill...

What I would like to do is to specify a waypoint with a certain distance/altitude to the earth's surface. So measured with a range sensor when the drone gets near the waypoint's lat, lon.

It is a pending idea/wish and Julian put some energy in it.
